paas_app_charmer/_gunicorn/wsgi_app.py

```diff
@@ -145,32 +145,43 @@ def map_integrations_to_env(integrations: IntegrationsState) -> dict[str, str]:
     Returns:
        A dictionary representing the environment variables for the IntegrationState.
     """
-    env = {}
+    env: dict[str, str | None] = {}
     if integrations.redis_uri:
         env["REDIS_DB_CONNECT_STRING"] = integrations.redis_uri
     for interface_name, uri in integrations.databases_uris.items():
         if uri is None:
             continue
         env_name = f"{interface_name.upper()}_DB_CONNECT_STRING"
         env[env_name] = uri
+
     if integrations.s3_parameters:
         s3 = integrations.s3_parameters
-        env_vars = {
-            "S3_ACCESS_KEY": s3.access_key,
-            "S3_SECRET_KEY": s3.secret_key,
-            "S3_REGION": s3.region,
-            "S3_STORAGE_CLASS": s3.storage_class,
-            "S3_BUCKET": s3.bucket,
-            "S3_ENDPOINT": s3.endpoint,
-            "S3_PATH": s3.path,
-            "S3_API_VERSION": s3.s3_api_version,
-            "S3_URI_STYLE": s3.s3_uri_style,
-            "S3_ADDRESSING_STYLE": s3.addressing_style,
-            "S3_ATTRIBUTES": json.dumps(s3.attributes) if s3.attributes else None,
-            "S3_TLS_CA_CHAIN": json.dumps(s3.tls_ca_chain) if s3.attributes else None,
-        }
-        for k, v in env_vars.items():
-            if v:
-                env[k] = v
-
-    return env
+        env.update(
+            {
+                "S3_ACCESS_KEY": s3.access_key,
+                "S3_SECRET_KEY": s3.secret_key,
+                "S3_REGION": s3.region,
+                "S3_STORAGE_CLASS": s3.storage_class,
+                "S3_BUCKET": s3.bucket,
+                "S3_ENDPOINT": s3.endpoint,
+                "S3_PATH": s3.path,
+                "S3_API_VERSION": s3.s3_api_version,
+                "S3_URI_STYLE": s3.s3_uri_style,
+                "S3_ADDRESSING_STYLE": s3.addressing_style,
+                "S3_ATTRIBUTES": json.dumps(s3.attributes) if s3.attributes else None,
+                "S3_TLS_CA_CHAIN": json.dumps(s3.tls_ca_chain) if s3.attributes else None,
+            }
+        )
+
+    if integrations.saml_parameters:
+        saml = integrations.saml_parameters
+        env.update(
+            {
+                "SAML_ENTITY_ID": saml.entity_id,
+                "SAML_METADATA_URL": saml.metadata_url,
+                "SAML_SINGLE_SIGN_ON_REDIRECT_URL": saml.single_sign_on_redirect_url,
```

Comment:
I miss the SingleLogoutUrl and both values can be either redirects or posts

Reply:
oops, that means we will need the `single_logout_service_redirect_binding`

Reply:
I will add the three missing fields

Reply:
After reviewing the spec, it is stated there that those fields will not be added yet. Good point anyway!

Reply:
Ok, no POST then, right?

Reply:
yes

Reply:
But as I understand it, we should have logout. Am I wrong?

Reply:
Nope, also stated in the spec `Single sign-out services are also optional and can be added in the future if necessary.`